akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -13,40 +13,65 @@ import akka.routing.ConsistentHashingRouter.ConsistentHashable
 import com.typesafe.config.Config
 import java.util.function.{ Function => JFunction }
 
+import akka.annotation.DoNotInherit
+import akka.io.dns.AAAARecord
+import akka.io.dns.ARecord
+import akka.io.dns.DnsProtocol
 import akka.util.unused
 import scala.collection.immutable
 
 import akka.event.Logging
 import akka.util.ccompat._
+import com.github.ghik.silencer.silent
 
+/**
+ * Not for user extension.
+ *
+ * This used to be a supported extension point but will be removed in future versions of Akka.
+ */
 @ccompatUsedUntil213
+@DoNotInherit
 abstract class Dns {
 
   /**
    * Lookup if a DNS resolved is cached. The exact behavior of caching will depend on
    * the akka.actor.io.dns.resolver that is configured.
    */
+  @deprecated("cached(DnsProtocol.Resolve)", "2.6.0")
   def cached(@unused name: String): Option[Dns.Resolved] = None
 
   /**
    * If an entry is cached return it immediately. If it is not then
    * trigger a resolve and return None.
    */
+  @deprecated("resolve(DnsProtocol.Resolve)", "2.6.0")
   def resolve(name: String)(system: ActorSystem, sender: ActorRef): Option[Dns.Resolved] = {
+    // doesn't delegate to new method as sender is expecting old protocol back
     val ret = cached(name)
     if (ret.isEmpty)
       IO(Dns)(system).tell(Dns.Resolve(name), sender)
     ret
   }
+
+  def cached(@unused request: DnsProtocol.Resolve): Option[DnsProtocol.Resolved] = None
+
+  def resolve(request: DnsProtocol.Resolve)(system: ActorSystem, sender: ActorRef): Option[DnsProtocol.Resolved] = {
```

Comment:
why second parameter list if not implicit parameters?

Reply:
Just copying the existing API, no need for this to be the same so i'll remove unless we think it would be better implicit? My thinking is not